src/Component/src/Metadata/Operation/HttpOperationInitiator.php

```diff
@@ -54,6 +56,33 @@ public function initializeOperation(Request $request): ?HttpOperation
             ->getOperation($metadata->getAlias(), $operationName)
         ;
 
-        return $operation;
+        return $this->getOperationWithVars($operation);
+    }
+
+    private function getOperationWithVars(HttpOperation $operation): HttpOperation
+    {
+        $operationVars = null !== $operation->getVars() ? $this->resolveVars($operation->getVars()) : null;
+
+        if (null !== $operationVars) {
+            $operation = $operation->withVars($operationVars);
+        }
+
+        $resource = $operation->getResource();
+        $resourceVars = $resource?->getVars();
+
+        if (null === $resourceVars) {
+            return $operation;
+        }
+
+        return $operation->withVars(\array_merge($this->resolveVars($resourceVars), $operationVars ?? []));
+    }
```

Comment:
Current method is a bit hard to read, would be to cool to make a bit more explicit.

For example,
```php
private function getOperationWithVars(HttpOperation $operation): HttpOperation
{
    $operationVars = $operation->getVars();
    $resolvedOperationVars = $operationVars !== null ? $this->resolveVars($operationVars) : null;

    $resourceVars = $operation->getResource()?->getVars();
    $resolvedResourceVars = $resourceVars !== null ? $this->resolveVars($resourceVars) : null;

    if ($resolvedOperationVars === null && $resolvedResourceVars === null) {
        return $operation;
    }

    $mergedVars = array_merge($resolvedResourceVars ?? [], $resolvedOperationVars ?? []);

    return $operation->withVars($mergedVars);
}
```